doc/devdocs/common/FilePreviewCommon.md

```diff
@@ -0,0 +1,70 @@
+# [FilePreviewCommon](/src/common/FilePreviewCommon)
+
+This project contains common code used for previewing and displaying files.
+
+## Monaco preview
+
+Monaco preview enables to display developer files. It is based on [Microsoft's Monaco Editor](https://microsoft.github.io/monaco-editor/) which is maintained by the Visual Studio Code team.
+
+This previewer is used for the File Explorer Dev File Previewer, as well as PowerToys Peek.
+
+### Update Monaco Editor
+
+1. Download Monaco editor with [npm](https://www.npmjs.com/): Run `npm i monaco-editor` in the command prompt.
+2. Delete everything except the `min` folder (the minimised code) from the downloaded files.
+3. Copy the `min` folder into the `src/common/FilePreviewCommon/monacoSRC` folder of the PowerToys project.
+4. Generate the JSON file as described in the generate [monaco_languages.json file](#monaco_languagesjson) section.
+
+### Add a new language definition
+
+As an example on how to add a new language definition you can look at the one for [registry files](/src/common/FilePreviewCommon/Assets/Monaco/customLanguages/reg.js).
+
+1. Add the language definition (written with [Monarch](https://microsoft.github.io/monaco-editor/monarch.html)) to the [folder containing Monaco custom languages](/src/common/FilePreviewCommon/Assets/Monaco/customLanguages/). The file should be formatted like in the example below. (Please change `idDefinition` to the name of your language.)
+
+```javascript
+export function idDefinition() {
+    return {
+        ...
+    }
+}
+```
+
+2. Add the following line to the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSpecialLanguages.js) file:
+
+```javascript
+import { idDefinition } from './customLanguages/file.js';
+```
+
+3. In the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalNewLanguage("id", [".fileExtension"], idDefinition(), monaco)
+```
+
+  * The id can be anything. Recommended is one of the file extensions. For example "php" or "reg".
+4. Copy the existing language definition into the `languageDefinitions` function in the same file. You can find the existing definitions in the following folder: [`/src/common/FilePreviewCommon//Assets/Monaco/monacoSRC/min/vs/basic-languages/`](/src/common/FilePreviewCommon/Assets/Monaco/monacoSRC/min/vs/basic-languages/).
```

Comment:
What should I copy why? Didn't I copied the definition file in an earlier step? Why should I copy something that exists in the target file?

Reply:
You're right. This should have been a step for the next section

Reply:
I read this document without doing the things and looking unto the existing files/folders. It is still not logical to me why yi should copy something that exists already? And you don't say exactly what I should copy?

It would make sense if I should copy an existing line to edit it. But this isn't what you wrote.

Reply:
Monaco needs it to display the original extension. I can't explain you exactly why but because how JavaScript works, I can't import these language definitions, so they have to be copied manually.